scripts/zones/Windurst_Waters/npcs/Ohbiru-Dohbiru.lua

```diff
@@ -221,38 +221,66 @@ function onEventFinish(player,csid,option)
-            player:delGil(250);
-            player:addItem(949); -- Rain Lily
-            player:messageSpecial(ITEM_OBTAINED,949);
-            player:needToZone(true);
-        elseif (option == 3) then
-            player:delGil(150);
-            player:addItem(956); -- Lilac
-            player:messageSpecial(ITEM_OBTAINED,956);
-            player:needToZone(true);
-        elseif (option == 4) then
-            player:delGil(200);
-            player:addItem(957); -- Amaryllis
-            player:messageSpecial(ITEM_OBTAINED,957);
-            player:needToZone(true);
-        elseif (option == 5) then
-            player:delGil(100);
-            player:addItem(958); -- Marguerite
-            player:messageSpecial(ITEM_OBTAINED,958);
-            player:needToZone(true);
-        elseif (option == 7) then
-            player:setVar("FLOWER_PROGRESS",1);
+        if (player:getFreeSlotsCount() > 0) then
+            if (option == 0) then
+                if (player:getGil() >= 300) then
+                    player:addItem(948); -- Carnation
+                    player:delGil(300);
+                    player:messageSpecial(ITEM_OBTAINED,948);
+                    player:needToZone(true);
+                else
+                    player:messageSpecial(NOT_HAVE_ENOUGH_GIL);
+                end
+            elseif (option == 1) then
+                if (player:getGil() >= 200) then
+                    player:addItem(941); -- Red Rose
+                    player:delGil(200);
+                    player:messageSpecial(ITEM_OBTAINED,941);
+                    player:needToZone(true);
+                else
+                    player:messageSpecial(NOT_HAVE_ENOUGH_GIL);
+                end
+            elseif (option == 2) then
+                if (player:getGil() >= 250) then
+                    player:addItem(949); -- Rain Lily
+                    player:delGil(250);
+                    player:messageSpecial(ITEM_OBTAINED,949);
+                    player:needToZone(true);
+                else
+                    player:messageSpecial(NOT_HAVE_ENOUGH_GIL);
+                end
+            elseif (option == 3) then
+                if (player:getGil() >= 150) then
+                    player:addItem(956); -- Lilac
+                    player:delGil(150);
+                    player:messageSpecial(ITEM_OBTAINED,956);
+                    player:needToZone(true);
+                else
+                    player:messageSpecial(NOT_HAVE_ENOUGH_GIL);
+                end
+            elseif (option == 4) then
+                if (player:getGil() >= 200) then
+                    player:addItem(957); -- Amaryllis
+                    player:delGil(200);
+                    player:messageSpecial(ITEM_OBTAINED,957);
+                    player:needToZone(true);
+                else
+                    player:messageSpecial(NOT_HAVE_ENOUGH_GIL);
+                end
+            elseif (option == 5) then
```

Comment:
not sure we want so many lines of repeated code
this whole tree can be put in a table
```lua
local tabre = 
{
    -- [key] = { itemid = id, gil = val },
    [5] = { itemid = 958, gil = 100 },
    ...
}
local choice = tabre[option]
if choice and player:getGil() >= choice.gil then
    player:addItem(choice.item)
    ...
end
```